swarm.go

```diff
@@ -352,17 +361,22 @@ func (s *Swarm) Start(srv *p2p.Server) error {
 	// update uaddr to correct enode
 	newaddr := s.bzz.UpdateLocalAddr([]byte(srv.Self().String()))
 	log.Info("Updated bzz local addr", "oaddr", fmt.Sprintf("%x", newaddr.OAddr), "uaddr", fmt.Sprintf("%s", newaddr.UAddr))
-	// set chequebook
-	//TODO: Currently if swap is enabled and no chequebook (or inexistent) contract is provided, the node would crash.
-	//Once we integrate back the contracts, this check MUST be revisited
-	if s.config.SwapEnabled && s.config.SwapAPI != "" {
+
+	// set up Swap
+	if s.config.SwapEnabled {
+		// check here again (maybe redundant): if enabled, we MUST have a contract API
+		if s.config.SwapAPI == "" {
+			return fmt.Errorf("Swap enabled but no contract address given; fatal error condition, aborting.")
```

Comment:
`fmt.Errorf` not needed. use `errors.New`

Reply:
fixed